### ISSUE TITLE
BatchNorm issue 

### ISSUE THREAD

**Issue by vishnureghu007** [BVLC/caffe#5829]
Hi
I have tried porting LinkNet (https://codeac29.github.io/projects/linknet/) using recent caffe.
I found that batch normalization doesnt works as expected.
In inference, modified use_global_stats: true, but the result is very weired. If use_global_stats: false, results looks somewhat to be good. It ran just 50 epochs.
I am just curious why BatchNorm layer affects in such a way.
Attached image with(true) and without(false)  use_global_stats .
![out copy](https://user-images.githubusercontent.com/15410424/29021777-2cf38b8a-7b56-11e7-8ce3-d3a00f2bcde2.png)
![out](https://user-images.githubusercontent.com/15410424/29021788-34ba9854-7b56-11e7-8cc1-b9cca75bf512.png)

**Comment by svobora** [comment]
First of all, do not use "use_global_stats" with the recent caffe version, it's done automagically.

**Comment by vishnureghu007** [comment]
without "use_global_stats" also gives the same problem

**Comment by shaibagon** [comment]
Please use the [caffe-users list][1] for usage, installation, or modeling questions, or other requests for help.
You may also post questions on [stackoverflow][3], make sure you tag them with `caffe` tag.  
There is also [caffe.help][4] documenting the different layers of caffe.
Do not post such requests to Issues. Doing so interferes with the development of Caffe.

Please read the [guidelines for contributing][2] before submitting this issue.

 [1]: https://groups.google.com/forum/#!forum/caffe-users
 [2]: https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md
 [3]: http://stackoverflow.com/questions/tagged/caffe
 [4]: http://caffe.help/manual/layers.html
